code/go/0chain.net/chaincore/chain/handler.go

```diff
@@ -1294,29 +1294,42 @@ func (c *Chain) N2NStatsWriter(w http.ResponseWriter, r *http.Request) {
 }
 
 /*PutTransaction - for validation of transactions using chain level parameters */
-func PutTransaction(ctx context.Context, entity datastore.Entity) (interface{}, error) {
-	txn, ok := entity.(*transaction.Transaction)
-	if !ok {
-		return nil, fmt.Errorf("put_transaction: invalid request %T", entity)
-	}
+func PutTransaction(c Chainer, lowerLevelFunc func(ctx context.Context, entity datastore.Entity) (interface{}, error)) datastore.JSONEntityReqResponderF {
+	return func(ctx context.Context, entity datastore.Entity) (interface{}, error) {
+		txn, ok := entity.(*transaction.Transaction)
+		if !ok {
+			return nil, fmt.Errorf("put_transaction: invalid request %T", entity)
+		}
 
-	sc := GetServerChain()
-	if sc.TxnMaxPayload() > 0 {
-		if len(txn.TransactionData) > sc.TxnMaxPayload() {
-			s := fmt.Sprintf("transaction payload exceeds the max payload (%d)", GetServerChain().TxnMaxPayload())
-			return nil, common.NewError("txn_exceed_max_payload", s)
+		if c.GetChainConfig().TxnMaxPayload() > 0 {
+			if len(txn.TransactionData) > c.GetChainConfig().TxnMaxPayload() {
+				s := fmt.Sprintf("transaction payload exceeds the max payload (%d)", c.GetChainConfig().TxnMaxPayload())
+				return nil, common.NewError("txn_exceed_max_payload", s)
+			}
 		}
-	}
 
-	// Calculate and update fee
-	if err := txn.ValidateFee(sc.ChainConfig.TxnExempt(), sc.ChainConfig.MinTxnFee()); err != nil {
-		return nil, err
-	}
-	if err := txn.ValidateNonce(); err != nil {
-		return nil, err
+		// Calculate and update fee
+		if err := txn.ValidateFee(c.GetChainConfig().TxnExempt(), c.GetChainConfig().MinTxnFee()); err != nil {
+			return nil, err
+		}
+		if err := txn.ValidateNonce(); err != nil {
+			return nil, err
+		}
+		/*
+			s, err := c.GetStateById(c.GetLatestFinalizedBlock().ClientState, c.GetChainConfig().OwnerID())
+			if !isValid(err) {
+				return nil, err
+			}
+			nonce := int64(0)
+			if s != nil {
+				nonce = s.Nonce
+			}
+			if txn.Nonce <= nonce {
+				return nil, errors.New("invalid transaction nonce")
+			}
+		*/
```

Comment:
The commented part is the code that we need, any reason we do use it.

Reply:
please rephrase in English

Reply:
To unit-test these changes, I needed to add this function signature to `Chainer` interface:
`GetStateById(clientState util.MerklePatriciaTrieI, clientID string) (*state.State, error)`
but this results in test build error
```
package 0chain.net/chaincore/chain
	imports 0chain.net/chaincore/mocks: import cycle not allowed in test
```
The same thing happens when adding a function signature having `config.ChainConfig` interface as output type.
`mockery` somehow doesn't allow to properly mock functions if they refer to interfaces defined in other packages,
though the main code is perfectly legal and works.
So what should we do about it?
Maybe we can move all the interfaces into a separate package like `interfaces` which would contain nothing but interfaces and this would help?
Or re-define `ChainConfig` and `MerklePatriciaTrieI` in the package `chaincore` as well?

Reply:
ping @peterlimg @dabasov

Reply:
@rrrooommmaaa please try mocks in package `0chain.net/chaincore/config/mocks`

Reply:
I need a mock for `Chainer`, with methods that return `config.ChainConfig`  and `util.MerklePatriciaTrieI`. This directory `0chain.net/chaincore/config/mocks` only contains the mock for `ChainConfig` interface.